### ISSUE TITLE
Own Dataset

### ISSUE THREAD

**Issue by sixftninja** [lwwu2/diver#2]
Hi, congratulations on your paper. The results look great and I can't wait to try it on my own data! Would it be possible for you to add instructions as to what is required?

**Comment by lwwu2** [comment]
Hi, thanks for your interest in our work.

Camera poses for images of your own data and the bounding box of the scene volume are required to train on our model. For the pose extraction, original [NeRF repo](https://github.com/bmild/nerf) has provided detailed instructions. And our code currently only supports cubic bounding box.

**Comment by Dev-MerB** [comment]
> Hi, thanks for your interest in our work.
> 
> Camera poses for images of your own data and the bounding box of the scene volume are required to train on our model. For the pose extraction, original [NeRF repo](https://github.com/bmild/nerf) has provided detailed instructions. And our code currently only supports cubic bounding box.

@lwwu2 First of all, congrats for your work! 
You said you followed the original nerf code as described above, but did you estimate the pose using colmap? In that case, how did you get the bbox and instrinct parameters?

I collected rendering images through Voreen, and also obtained GT camera pose values and focal lengths (opengl format transform matrix). The direction the camera is looking in doesn't match exactly, so I'm wondering how to fix this

Here is [my data and config settings.](https://drive.google.com/file/d/1C2h9R7k0m6pSzE8A8XKNr2AzL-_8tqEQ/view?usp=sharing) Can you please check it out?

![image](https://user-images.githubusercontent.com/39505929/182080652-f00bdb07-f39b-4d9f-9f2d-bf7a24b99d78.png)


+) Additionally, if you try to train with that data, the VRAM only consumes about 3GB, and if you use other data from NSVF (Statues), it occupies 15GB.

**Comment by lwwu2** [comment]
Hi, thanks for your interest in our work.

The evaluation dataset (NeRF-synthetic, TnT, BlendedMVS) used in our paper already have pose, bbox, and intrinsic parameters provided. So, we did not re-estimate them.  Colmap pose estimation is needed if you need to train on real world scenes. If the scene is not normalized to be within the range of [-1,1], the bbox should also be estimated from things like the point cloud or be marked manually.

We hard code focal length and some data loading logics in both dataloader. So you might need to modify the code to match your dataset. If I remember correctly, the transform matrix used in dataloader of TnT and BlendedMVS is not in OpenGL format, which might be the reason why the direction looks not correct. Meanwhile, since your ground truth image has white background, you might also need to set 'white_back' to True in 'config.py'.

**Comment by Dev-MerB** [comment]
> Hi, thanks for your interest in our work.
> 
> The evaluation dataset (NeRF-synthetic, TnT, BlendedMVS) used in our paper already have pose, bbox, and intrinsic parameters provided. So, we did not re-estimate them. Colmap pose estimation is needed if you need to train on real world scenes. If the scene is not normalized to be within the range of [-1,1], the bbox should also be estimated from things like the point cloud or be marked manually.
> 
> We hard code focal length and some data loading logics in both dataloader. So you might need to modify the code to match your dataset. If I remember correctly, the transform matrix used in dataloader of TnT and BlendedMVS is not in OpenGL format, which might be the reason why the direction looks not correct. Meanwhile, since your ground truth image has white background, you might also need to set 'white_back' to True in 'config.py'.

Thanks for your reply.

I have an additional question.
As described in the original nerf repo, I used imgs2poses.py to get data in LLFF format (poses_bounds.npy). Can you provide a dataloader that uses that data format?? Or do you know the process of converting the data to json format?

**Comment by lwwu2** [comment]
> Thanks for your reply.
> 
> I have an additional question. As described in the original nerf repo, I used imgs2poses.py to get data in LLFF format (poses_bounds.npy). Can you provide a dataloader that uses that data format?? Or do you know the process of converting the data to json format?

Hi,

I think the conversion can be done by reading the pose in the npy file as the camera extrinsic matrix then saving it in json format as 'transform_matrix', together with the 'file_path' corresponds to the image path. The 'rotation' can be ignored since it is never used and you might need to convert the extrinsic matrix to opengl format.

**Comment by Dev-MerB** [comment]
> > Thanks for your reply.
> > I have an additional question. As described in the original nerf repo, I used imgs2poses.py to get data in LLFF format (poses_bounds.npy). Can you provide a dataloader that uses that data format?? Or do you know the process of converting the data to json format?
> 
> Hi,
> 
> I think the conversion can be done by reading the pose in the npy file as the camera extrinsic matrix then saving it in json format as 'transform_matrix', together with the 'file_path' corresponds to the image path. The 'rotation' can be ignored since it is never used and you might need to convert the extrinsic matrix to opengl format.

Thank you for the information

**Comment by yuemingPAN** [comment]
hello, thanks for your great work ,I also have a qusention. if the transform matrix used in dataloader of TnT and BlendedMVS is not in OpenGL format, is it in OpenCV farmat or other farmat? look forward to your reply. Thanks

**Comment by lwwu2** [comment]
Hi,

I think it should be in the original format of BlendedMVS dataset https://github.com/YoYo000/MVSNet

**Comment by yuemingPAN** [comment]
Thanks for your reply! I want to ask another question, what's the meaning of this line of code https://github.com/lwwu2/diver/blob/78684e5e5004896be9bc1cc3eaa01bb94f02c042/utils/dataset/tanks.py#L180
It seems that the second and third columns of the extrinsic matrix are taken as the original opposite numbers. Does it mean that the y and z axes are reversed? After this line of code, does the extrinsic matrix conform to the opengl coordinate system?
I'm a new learner, sorry if the question is trivial.

**Comment by lwwu2** [comment]
Yes, it reverses the y and z axes and converts to the opengl coordinate system.
